realm/realm-library/src/androidTest/java/io/realm/RxJavaTests.java

```diff
@@ -557,42 +559,42 @@ public void call(Realm rxRealm) {
     public void dynamicRealm_closeInDoOnUnsubscribe() {
         final DynamicRealm dynamicRealm = DynamicRealm.getInstance(realm.getConfiguration());
 
-        Observable<DynamicRealm> observable = dynamicRealm.asObservable()
-                .doOnUnsubscribe(new Action0() {
+        Flowable<DynamicRealm> observable = dynamicRealm.asFlowable()
+                .doOnCancel(new Action() {
                     @Override
-                    public void call() {
+                    public void run() throws Exception {
                         dynamicRealm.close();
                     }
                 });
 
-        subscription = observable.subscribe(new Action1<DynamicRealm>() {
+        subscription = observable.subscribe(new Consumer<DynamicRealm>() {
             @Override
-            public void call(DynamicRealm rxRealm) {
+            public void accept(DynamicRealm ignored) throws Exception {
             }
         });
 
-        subscription.unsubscribe();
+        subscription.dispose();
         assertTrue(dynamicRealm.isClosed());
     }
 
     @Test
     @UiThreadTest
     public void realmResults_closeInDoOnUnsubscribe() {
```

Comment:
It seems to me that `doOnUnsubscribe` is more accurate, perhaps `doOnDispose`? From what I can tell `unsubscribe` is the general word used for telling a stream that you no longer want it around, you can do that by either calling `dispose`, not `cancel`. Also, subscribing returns a `Disposable`, not a `Cancellable`